azure-pipelines.yml

```diff
@@ -0,0 +1,152 @@
+    displayName: install python basics
+  - script: |
+      mkdir $(venv)
+      $(buildPython.pythonLocation)\python -m virtualenv $(venv)
+    displayName: setup virtual environment
+  - ${{ if eq(variables.System.PullRequest.TargetBranch, 'develop') }}:
+    - powershell: |
+        $ver = (git show -s --format=%cd --date=format:'%Y%m%d%H%M%S')
+        (Get-Content setup.cfg) -replace '^(version = )(\d+\.\d+\.\d+).*$', ('$1$2.dev' + $ver) | Set-Content setup.cfg
+      displayName: mark the version as development
+  - script: |
+      call $(venv)\Scripts\activate.bat
+      python -m pip install -r requirements\pip.txt
+      python -m pip install -r requirements\build.txt
+      pip install --editable .
+      gravitybee --src-dir src --sha file --with-latest --extra-data static --extra-pkgs boto3 --extra-modules boto3
+    displayName: create standalone application
+  - script: |
+      call .gravitybee\gravitybee-environs.bat
+      mkdir $(deployStage)
+      move %GB_ENV_STAGING_DIR% $(deployStage)
+      move .gravitybee\gravitybee-environs.bat $(deployStage)
+      move requirements\deploy.txt $(deployStage)
+    displayName: prepare deploy
+  - script: dir $(deployStage)
+    displayName: see directory
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathtoPublish: $(deployStage)
+      artifactName: $(deployArtifactName)
+    displayName: publish standalone artifacts
+
+- deployment: deploy
+  displayName: deploy standalone to github and s3
+  dependsOn: build
+  condition: succeeded()
+  pool:
+    vmImage: 'vs2017-win2016'
+  environment: 'standalone'
+  variables: # alternate synatx needed when using a group
+  - group: wam-windows-ci
+  - name: deployStage
+    value: $(System.DefaultWorkingDirectory)\$(deployStageDir)
+  - name: keyPrefix
+    value: test_releases
+  strategy:
+    runOnce:
+      deploy:
+        steps:
+        - checkout: none
+        - task: DownloadBuildArtifacts@0
+          displayName: download build artifacts
+          inputs:
+            artifactName: $(deployArtifactName)
+            downloadPath: $(deployStage)
+        - task: S3Upload@1
+          displayName: upload dev standalone binaries to s3 (dev)
+          condition: eq(variables['build.sourceBranch'], 'refs/heads/develop')
+          inputs:
+            awsCredentials: dicelab
+            targetFolder: $(keyPrefix)
+            #regionName: 'us-east-1'
+            bucketName: $(devBucket)
+            sourceFolder: $(deployStage)
+        - task: S3Upload@1
+          displayName: upload standalone binaries to s3 (release)
+          condition: eq(variables['build.sourceBranch'], 'refs/tags/*')
+          inputs:
+            awsCredentials: dicelab
+            targetFolder: $(keyPrefix)
+            #regionName: 'us-east-1'
+            bucketName: $(releaseBucket)
+            sourceFolder: $(deployStage)
+        - script: |
```

Comment:
Do we know that non-zero return codes from individual commands in these multi-line script-blocks will cause the script-block to fail? Or, like in bash or powershell by default, will it continue on to the next command, potentially causing a false success or otherwise obscuring the original error?